dotnet/src/Connectors/Connectors.OpenAI/OpenAIServiceCollectionExtensions.cs

```diff
@@ -213,14 +213,124 @@ public static class OpenAIServiceCollectionExtensions
                 serviceProvider.GetService<ILoggerFactory>()));
     }
 
+    /// <summary>
+    /// Adds an Azure OpenAI text generation service with the specified configuration.
+    /// </summary>
+    /// <param name="builder">The <see cref="IKernelBuilder"/> instance to augment.</param>
+    /// <param name="config">Required configuration for Azure OpenAI.</param>
+    /// <param name="httpClient">The HttpClient to use with this service.</param>
+    /// <returns>The same instance as <paramref name="builder"/>.</returns>
+    [Experimental("SKEXP0010")]
+    public static IKernelBuilder AddAzureOpenAITextGeneration(
+        this IKernelBuilder builder,
+        AzureOpenAIApiKeyConfig config,
+        HttpClient? httpClient = null)
+    {
+        Verify.NotNull(builder);
+        Verify.NotNull(config);
+
+        config.Validate();
+
+        builder.AddAzureOpenAITextGeneration(
+            deploymentName: config.Deployment,
+            endpoint: config.Endpoint,
+            apiKey: config.ApiKey,
+            serviceId: config.ServiceId,
+            modelId: config.Deployment,
+            httpClient: httpClient);
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds an Azure OpenAI text generation service with the specified configuration.
+    /// </summary>
+    /// <param name="builder">The <see cref="IKernelBuilder"/> instance to augment.</param>
+    /// <param name="config">Required configuration for Azure OpenAI.</param>
+    /// <param name="httpClient">The HttpClient to use with this service.</param>
+    /// <returns>The same instance as <paramref name="builder"/>.</returns>
+    [Experimental("SKEXP0010")]
+    public static IKernelBuilder AddAzureOpenAITextGeneration(
+        this IKernelBuilder builder,
+        AzureOpenAITokenConfig config,
+        HttpClient? httpClient = null)
+    {
+        Verify.NotNull(builder);
+        Verify.NotNull(config);
+
+        config.Validate();
+
+        builder.AddAzureOpenAITextGeneration(
+            deploymentName: config.Deployment,
+            endpoint: config.Endpoint,
+            credentials: config.TokenCredential!,
+            serviceId: config.ServiceId,
+            modelId: config.Deployment,
```

Comment:
In current extension methods `deploymentName` and `modelId` properties are separate, same as on Azure side. We should keep that logic with new configuration models as well.